WcaOnRails/app/webpacker/components/Delegates/index.jsx

```diff
@@ -77,94 +86,136 @@ function DelegatesOfRegion({ activeSeniorDelegate, delegates, isAdminMode }) {
             <Table.Cell>
               {I18n.t(`enums.user.delegate_status.${delegate.delegate_status}`)}
             </Table.Cell>
-            <Table.Cell>
-              {delegate.location}
-            </Table.Cell>
+            <Table.Cell>{delegate.location}</Table.Cell>
           </Table.Row>
         ))}
       </Table.Body>
     </Table>
   );
 }
 
-export default function Delegates({
-  delegates,
-  isEditVisible,
-}) {
-  const seniorDelegates = React.useMemo(() => delegates
-    .filter((user) => user.delegate_status === 'senior_delegate')
-    .sort((user1, user2) => (user1.location || '').localeCompare(user2.location || '')), [delegates]);
+export default function Delegates() {
+  const { data, loading, error } = useLoadedData(delegatesStaticPageDataUrl);
+
+  const { delegates, canViewDelegateMatters } = data || {};
+  const seniorDelegates = React.useMemo(() => {
+    if (delegates) {
+      return delegates
+        .filter((user) => user.delegate_status === 'senior_delegate')
+        .sort((user1, user2) => (user1.location || '').localeCompare(user2.location || ''));
+    }
+    return null;
+  }, [delegates]);
+
+  const [activeSeniorDelegate, setActiveSeniorDelegate] = React.useState();
+  const [adminMode, setAdminMode] = React.useState(false);
 
-  const [activeSeniorDelegate, setActiveSeniorDelegate] = React.useState(seniorDelegates?.[0]);
+  React.useEffect(() => {
+    setActiveSeniorDelegate(seniorDelegates?.[0]);
+  }, [seniorDelegates]);
```

Comment:
Why are you assuming that there are multiple Senior Delegates, and then "hooking in" a single Senior Delegate? I am asking because this smells like a bad way to use the `useEffect` hook.

Reply:
The `seniorDelegates` array has the list of all senior delegates (from all regions). This useEffect is making the first senior delegate as the 'selected' senior delegate. Later, when user switches between regions, it will change again. Also, this is temporary because we will soon migrate to the user_groups API to fetch the regions and roles API to fetch the delegates. Once the APIs are ready, we will be able to migrate it to a better way.

Reply:
Okay, then I'll accept it as a "temporary thing" :upside_down_face: